qi-doc/scribblings/principles.scrbl

```diff
@@ -122,3 +122,20 @@ It turns out that the core routing forms of Qi fulfill the definition of @hyperl
 So evidently, flows are just @hyperlink["https://www.sciencedirect.com/science/article/pii/S1571066106001666/pdf"]{monoids in suitable subcategories of bifunctors} (what's the problem?), or, in another way of looking at it, @hyperlink["https://bentnib.org/arrows.pdf"]{enriched Freyd categories}.
 
 Therefore, any theoretical results about arrows should generally apply to Qi as well (but not necessarily, since Qi is not @emph{just} arrows).
+
+@section{It's Languages All the Way Down}
+
+Qi is a language implemented on top of another language, Racket, by means of a macro called @racket[flow]. All of the other macros that serve as Qi's @seclink["Embedding_a_Hosted_Language"]{embedding} into Racket, such as (the Racket macros) @racket[~>] and @racket[switch], expand to a use of @racket[flow].
+
+@racket[flow] accepts Qi syntax and (like any @tech/reference{macro}) produces Racket syntax. It does this in two stages:
```

Comment:
In mathematical writing, it's usually considered bad form to start a sentence with a variable or formula, like in "x must be non-zero." A possible rewording:
```suggestion
The @racket[flow] form accepts Qi syntax and (like any @tech/reference{macro}) produces Racket syntax. It does this in two stages:
```